### PR TITLE
Rename pbmesh.Upstreams to pbmesh.Destinations

### PR DESCRIPTION
**Pull request by ishustava**

### CODE REVIEW

File: control-plane/connect-inject/controllers/pod/pod_controller.go
Comment (wilkermichael) [inline control-plane/connect-inject/controllers/pod/pod_controller.go:1]: Should the comment here be `Destination Resources`

```
// of the Pod's Status field. Controller is also responsible for generating V2 Upstreams resources
```

File: control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go [hunk 1]
```diff
@@ -330,7 +330,7 @@ func TestReconcileDeletePodWithMirrorNamespaces(t *testing.T) {
 			existingWorkload:           createWorkload(),
 			existingHealthStatus:       createPassingHealthStatus(),
 			existingProxyConfiguration: createProxyConfiguration(testPodName, pbmesh.ProxyMode_PROXY_MODE_DEFAULT),
-			existingUpstreams:          createUpstreams(),
```
Comment (wilkermichael) [inline control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go:333]: should be `existingDestinations:`. Just double check for other instances of upstream in this and other files

File: control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func TestReconcileCreatePodWithMirrorNamespaces(t *testing.T) {
 			expectedWorkload:           createWorkload(),
 			expectedHealthStatus:       createPassingHealthStatus(),
 			expectedProxyConfiguration: createProxyConfiguration(testPodName, pbmesh.ProxyMode_PROXY_MODE_DEFAULT),
-			expectedUpstreams:          createUpstreams(),
```
Comment (wilkermichael) [inline control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go:197]: should be `existingDestinations:`

**Review by wilkermichael — APPROVED** [review]
Thanks for refactoring Iryna!

Just some missed `upstreams`

File: control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go (hunk 2 above)
Comment (ishustava) [reply control-plane/connect-inject/controllers/pod/pod_controller_ent_test.go:197]: Yep, looks like I forgot the ent test file!

**Pull request merged**